### PR TITLE
modified device selection

### PR DESCRIPTION
**Pull request by Stephenfang51**
device cannot sucessfully control through argments "device"

Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily receiving feedbacks. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Please describe the motivation of this PR and the goal you want to achieve through this PR.

## Modification

Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repositories?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit tests to ensure the correctness.
3. If the modification has a dependency on downstream projects of a newer version, this PR should be tested with all supported versions of downstream projects.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

**Review by AllentDan — APPROVED** [review]
LGTM

**Review by RunningLeon — DISMISSED** [review]
LGTM

**Comment by RunningLeon** [comment]
@Stephenfang51 Hi, could you fix lint?

**Comment by Stephenfang51** [comment]
> @Stephenfang51 Hi, could you fix lint?

Sure, I will take a look at this. Not familiar with the lint actually.

File: mmdeploy/backend/onnxruntime/wrapper.py [hunk 1]
```diff
@@ -50,18 +50,13 @@ def __init__(self,
...
         is_cuda_available = ort.get_device() == 'GPU'
-        if is_cuda_available:
-            providers.insert(0, 'CUDAExecutionProvider')
-            options.insert(0, {'device_id': device_id})
-        sess.set_providers(providers, options)
+        providers = [('CUDAExecutionProvider', {'device_id': device_id})] \
+            if is_cuda_available else ['CPUExecutionProvider']
+        sess = ort.InferenceSession(onnx_file, session_options, providers=providers)
+
+        # sess.set_providers(providers, options)
```
Comment (AllentDan) [inline mmdeploy/backend/onnxruntime/wrapper.py:59]: may get removed if useless

**Review by AllentDan — COMMENTED** [review]
Hi, @Stephenfang51. Could you fix the lint first?

File: mmdeploy/backend/onnxruntime/wrapper.py (hunk 1 above)
Comment (Stephenfang51) [reply mmdeploy/backend/onnxruntime/wrapper.py]: hope I got this right this time.

**Review by AllentDan — APPROVED** [review]

**Review by RunningLeon — APPROVED** [review]
LGTM

**Comment by OpenMMLab-gd** [comment]
Dear Stephenfang51,
First of all, we want to express our gratitude for your significant PR in the MMDeploy project. Your contribution is highly appreciated, and we are grateful for your efforts in helping improve this open-source project during your personal time. We believe that many developers will benefit from your PR.
If you are Chinese or have WeChat，welcome to join our community on WeChat. You can add our assistant ：openmmlabwx. Please add "mmsig + Github ID" as a remark when adding friends：）
We would also like to invite you to join our Special Interest Group (SIG) private channel on Discord, where you can share your experiences, ideas, and build connections with like-minded peers. To join the SIG channel, simply message moderator— OpenMMLab on Discord or briefly share your open-source contributions in the #introductions channel and we will assist you. We look forward to seeing you there! Join us ：https://discord.gg/raweFPmdzG
Thank you again for your contribution❤
Best regards！ @Stephenfang51

**Pull request merged**